spec/features/posts_index_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+describe 'Users Post Index', type: :feature do
+  before :each do
+    @user = User.create(name: 'Muhammed-ali', photo: 'https://Akaiiii.jpg', bio: 'Martial Artist')
+    @user2 = User.create(name: 'Bonzai73', photo: 'https://bon-bon.jpg', bio: 'Monk of the mountains')
+    @post1 = Post.create(author_id: @user.id, title: 'The way of the Water',
+                         text: 'In martial arts, the way of the water is the way of the soft and yielding')
+    @post2 = Post.create(author_id: @user.id, title: 'Amaterasu',
+                         text: 'Amaterasu is the goddess of the sun and the universe, the ultimate source of all life')
+    @post3 = Post.create(author_id: @user.id, title: 'Legacy', text: 'The legacy of the righteous is a blessing')
+    Comment.create(user_id: @user2.id, post_id: @post1.id, text: 'I like this post')
+    Comment.create(user_id: @user.id, post_id: @post1.id, text: 'Thanks! I wrote it myself based on my experiences')
+    Like.create(user_id: @user2.id, post_id: @post1.id)
+    Like.create(user_id: @user.id, post_id: @post1.id)
+    visit user_posts_path(@user.id)
+  end
+
+  scenario "I can see the user's profile picture" do
+    expect(page).to have_css("img[src*='https://Akaiiii.jpg']")
+  end
+
+  scenario "I can see the user's username" do
+    expect(page).to have_content('Muhammed-ali')
+  end
+
+  scenario 'I can see the number of posts the user has written' do
+    expect(page).to have_content('Number of posts: 3')
+  end
+
+  scenario "I can see a post's title" do
+    expect(page).to have_content('The way of the Water')
+  end
+
+  scenario "I can see some of the post's body" do
+    expect(page).to have_content('In martial arts, the way of the water is the way of the soft and yielding')
+  end
+
+  scenario 'I can see the first comments on a post' do
+    expect(page).to have_content('I like this post')
+  end
+
+  scenario 'I can see how many comments a post has' do
+    expect(page).to have_content('Comments: 2')
+  end
+
+  scenario 'I can see how many likes a post has' do
+    expect(page).to have_content('Likes: 2')
+  end
+
+  scenario "When I click on a post, it redirects me to that post's show page" do
+    click_on('The way of the Water')
+    expect(page).to have_content('In martial arts, the way of the water is the way of the soft and yielding')
+  end
```

Comment:
- [ ] As mentioned by the previous reviewer you forgot to test the pagination, there should be a pagination button so if there are more posts that fit on the view we can click on the pagination button to see the other post. Kindly test the pagination as well 🙃 🙃 
- [ ] Also the test for `when clicking on a post, it redirects to that post's show page`, means you should test if when the user clicks on the post he is directed to the post-show path, it should show the content of all the page and not the description only. Instead of having the post description you should have the post show page link 😁

Reply:
@cynthiainga I added that in the **users_show_spec.rb** file

Reply:
Hello @matee-safi, you are supposed to put the test for pagination inside the `User post index page` since it is a project requirement.

Reply:
Hello @matee-safi, that one may be for the `See all post` button, but you need to add a pagination test for the post index page since it is part of the project requirements.